tests/storage/export.go

```diff
@@ -2370,6 +2373,108 @@ var _ = SIGDescribe("Export", func() {
+					vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, &metav1.GetOptions{})
+					if errors.IsNotFound(err) {
+						return ""
+					}
+					Expect(err).ToNot(HaveOccurred())
+					return vmi.Status.Phase
+				}, 180*time.Second, time.Second).Should(Equal(virtv1.Running))
+
+				By("Stopping VM, we should get the export ready eventually")
+				vm = stopVM(vm)
+
+				// Run vmexport
+				By("Running vmexport create command")
+				virtctlCmd := clientcmd.NewRepeatableVirtctlCommand(commandName,
+					"create",
+					vmeName,
+					"--vm", vm.Name,
+					"--namespace", testsuite.GetTestNamespace(vm))
+				err = virtctlCmd()
+				Expect(err).ToNot(HaveOccurred())
+
+				exportReady := exportv1.Ready
+				Eventually(func() *exportv1.VirtualMachineExportPhase {
+					vme, err := virtClient.VirtualMachineExport(vm.Namespace).Get(context.Background(), vmeName, metav1.GetOptions{})
+					if errors.IsNotFound(err) {
+						return nil
+					}
+					Expect(err).ToNot(HaveOccurred())
+					if vme.Status == nil {
+						return nil
+					}
+					return &vme.Status.Phase
+				}, 180*time.Second, time.Second).Should(Equal(&exportReady))
+				vme, err := virtClient.VirtualMachineExport(vm.Namespace).Get(context.Background(), vmeName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				cmdArgs := []string{
+					commandName,
+					"manifest",
+					vmeName,
+					"--namespace",
+					testsuite.GetTestNamespace(vm),
+				}
+				if vme.Status.Links.External == nil {
+					targetPort := 37548 + rand.Intn(6000)
+					_, forwardCmd, err = clientcmd.CreateCommandWithNS(vm.Namespace, clientcmd.GetK8sCmdClient(),
+						"port-forward", fmt.Sprintf("service/virt-export-%s", vmeName), fmt.Sprintf("%d:443", targetPort))
+					Expect(err).ToNot(HaveOccurred())
+					go func() {
+						defer GinkgoRecover()
+						Expect(err).ToNot(HaveOccurred())
+						stdout, err := forwardCmd.StdoutPipe()
+						Expect(err).ToNot(HaveOccurred())
+						err = forwardCmd.Start()
+						Expect(err).ToNot(HaveOccurred())
+						waitForPortForwardCmd(stdout, 443, targetPort)
+					}()
+
+					cmdArgs = append(cmdArgs, "--export-url", fmt.Sprintf("127.0.0.1:%d", targetPort),
+						"--insecure")
+				}
+				cmdArgs = append(cmdArgs, extraArgs...)
+				virtctlCmdOut := clientcmd.NewRepeatableVirtctlCommandWithOut(cmdArgs...)
+
+				var out []byte
+				By("Running vmexport manifest command")
+				Eventually(func() error {
+					out, err = virtctlCmdOut()
+					return err
+				}, 30*time.Second, time.Second).Should(BeNil())
+				Expect(out).NotTo(BeEmpty())
+				split := strings.Split(string(out), "\n---\n")
+				// Add one for the --- at the end.
+				Expect(split).To(HaveLen(expectedObjects + 1))
+				resVM := &virtv1.VirtualMachine{}
+				err = yaml.Unmarshal([]byte(split[1]), resVM)
+				Expect(err).ToNot(HaveOccurred())
+			},
+				Entry("without --include-secret", 2),
+				Entry("with --include-secret", 3, "--include-secret"),
```

Comment:
with/without export-url?

Should we test export-url with volume test as well?

Reply:
Well this is `with` export-url so I can do the port-forward. If I do without, I would need a properly configured Ingress/Route, and I would have to skip if I don't have such a thing. And the entire reason I added `--export-url` is because I don't have a properly configured Ingress/Route. As you noted the volume export tests are all skipped because we don't have an Ingress/Route here.

Reply:
I actually didn't know that the volume export tests are skipped.  Can we do them now?

Reply:
If you check the latest push on this PR, I modified some of the existing ones to use port forward (Not all because some of them use `create and download` the image, and there is no chance to port forward in that process). Once we get the --port-forward flag implemented we can convert them as well. There should be comments indicating this.